core/ledger/kvledger/benchmark/chainmgmt/chains.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package chainmgmt
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/common/configtx/test"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/ledger/ledgermgmt"
+	"github.com/hyperledger/fabric/core/ledger/ledgermgmt/ledgermgmttest"
+	"github.com/hyperledger/fabric/core/ledger/util/couchdb"
+)
+
+// ChainID is a type used for the ids for the chains for experiments
+type ChainID int
+
+func (chainID ChainID) String() string {
+	return fmt.Sprintf("%s%04d", "chain-", chainID)
+}
+
+// SimulationResult is a type used for simulation results
+type SimulationResult []byte
+
+// chainsMgr manages chains for experiments
+type chainsMgr struct {
+	ledgerMgr *ledgermgmt.LedgerMgr
+	mgrConf   *ChainMgrConf
+	batchConf *BatchConf
+	initOp    chainInitOp
+	chainsMap map[ChainID]*Chain
+	wg        *sync.WaitGroup
+}
+
+func newChainsMgr(mgrConf *ChainMgrConf, batchConf *BatchConf, initOp chainInitOp) *chainsMgr {
+	dataDir := filepath.Join(mgrConf.DataDir, "ledgersData")
+	ledgermgmtInitializer := ledgermgmttest.NewInitializer(dataDir)
+	ledgermgmtInitializer.Config.HistoryDBConfig.Enabled = true
+	if os.Getenv("useCouchDB") == "true" {
+		ledgermgmtInitializer.Config.StateDBConfig.StateDatabase = "CouchDB"
+		ledgermgmtInitializer.Config.StateDBConfig.CouchDB = &couchdb.Config{
+			Address:            "localhost:5984",
+			RedoLogPath:        filepath.Join(dataDir, "couchdbRedologs"),
+			UserCacheSizeMBs:   500,
+			MaxBatchUpdateSize: 500,
+		}
+	}
```

Comment:
Seems that we should be using the same patterns that we use in the rest of the tests and avoid hard-coded hosts and ports of cloud and rely on what's in COUCHDB_ADDR.